Spigot-Server-Patches/0167-Anti-Xray.patch

```diff
@@ -0,0 +1,1519 @@
+From a1200045bc1972a1f4ebf69f3702d1734a544fce Mon Sep 17 00:00:00 2001
+From: stonar96 <minecraft.stonar96@gmail.com>
+Date: Tue, 30 Aug 2016 00:49:26 +0200
+Subject: [PATCH] Anti-Xray
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index e3fe8f4..828812d 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -1,5 +1,6 @@
+ package com.destroystokyo.paper;
+ 
++import java.util.Arrays;
+ import java.util.List;
+ 
+ import org.bukkit.Bukkit;
+@@ -382,4 +383,26 @@ public class PaperWorldConfig {
+     private void elytraHitWallDamage() {
+         elytraHitWallDamage = getBoolean("elytra-hit-wall-damage", true);
+     }
++
++    public boolean antiXray;
++    public boolean asynchronous;
++    public int engineMode;
++    public int neighborsMode;
++    public int maxChunkY;
++    public List<Object> hiddenBlocks;
++    public List<Object> replaceBlocks;
++    private void antiXray() {
++        antiXray = getBoolean("anti-xray.enabled", false);
++        asynchronous = true;
++        engineMode = getInt("anti-xray.engine-mode", 3);
++        neighborsMode = getInt("anti-xray.neighbors-mode", 1);
++        maxChunkY = getInt("anti-xray.max-chunk-y", 3);
++        hiddenBlocks = getList("anti-xray.hide-blocks", Arrays.asList(new Object[] {
++            "gold_ore", "iron_ore", "coal_ore", "lapis_ore", "mossy_cobblestone", "obsidian", "chest", "diamond_ore", "redstone_ore", "lit_redstone_ore", "clay", "emerald_ore", "ender_chest"
++        }));
++        replaceBlocks = getList("anti-xray.replace-blocks", Arrays.asList(new Object[] {
```

Comment:
`new Object[]` should not be necessary here: `Arrays.astList("stone", "planks")`.